[AArch64] Guard against non-simple types in udiv sve costs.

davemgreen opened the pull request:
The code here probably needs to change to handle types more uniformly, but this prevents it from trying to use a simple type where it does not exist.

Fixes #148438.

c-rhodes commented on llvm/test/CodeGen/AArch64/saturating-vec-smull.ll:
think you've added this by mistake?

c-rhodes reviewed (approved):
LGTM cheers, minus the query on the new test you've added

davemgreen committed:
[AArch64] Guard against non-simple types in udiv sve costs.

The code here probably needs to change to hand;e types moe uniformly, but this
prevents it from trying to use a simple type where that does not exist.

rj-jesus reviewed (approved):
LGTM, thanks!
